docs/scripts/generate-termynal.py

```diff
@@ -0,0 +1,139 @@
+import shutil
+from pathlib import Path
+
+import pexpect
+from ansi2html import Ansi2HTMLConverter
+
+CCDS_ROOT = Path(__file__).parents[2].resolve()
+
+
+def execute_command_and_get_output(command, input_script):
+    input_script = iter(input_script)
+    child = pexpect.spawn(command, encoding="utf-8")
+
+    interaction_history = [f"$ {command}\n"]
+
+    prompt, user_input = next(input_script)
+
+    try:
+        while True:
+            index = child.expect([prompt, pexpect.EOF, pexpect.TIMEOUT])
+
+            if index == 0:
+                output = child.before + child.after
+                interaction_history += [line.strip() for line in output.splitlines()]
+
+                child.sendline(user_input)
+
+                try:
+                    prompt, user_input = next(input_script)
+                except StopIteration:
+                    pass
+
+            elif index == 1:  # The subprocess has exited.
+                output = child.before
+                interaction_history += [line.strip() for line in output.splitlines()]
+                break
+            elif index == 2:  # Timeout waiting for new data.
+                print("\nTimeout waiting for subprocess response.")
+                continue
+
+    finally:
+        return interaction_history
+
+
+ccds_script = [
+    ("project_name", "My Analysis"),
+    ("repo_name", "my_analysis"),
+    ("module_name", ""),
+    ("author_name", "Dat A. Scientist"),
+    ("description", "This is my analysis of the data."),
+    ("python_version_number", "3.12"),
+    ("Choose from", "3"),
+    ("bucket", "s3://my-aws-bucket"),
+    ("aws_profile", ""),
+    ("Choose from", "2"),
+    ("Choose from", "1"),
+    ("Choose from", "2"),
+    ("Choose from", "2"),
+    ("Choose from", "1"),
+]
+
+
+def run_scripts():
+    try:
+        output = []
+        output += execute_command_and_get_output(f"ccds {CCDS_ROOT}", ccds_script)
+        return output
+
+    finally:
+        # always cleanup
+        if Path("my_analysis").exists():
+            shutil.rmtree("my_analysis")
+
+
+def render_termynal():
+    # actually execute the scripts and capture the output
+    results = run_scripts()
+
+    # watch for inputs and format them differently
+    script = iter(ccds_script)
+    _, user_input = next(script)
+
+    conv = Ansi2HTMLConverter(inline=True)
+    html_lines = [
+        '<div id="termynal" data-termynal class="termy" data-ty-lineDelay="100" data-ty-typeDelay="50" title="Cookiecutter Data Science">'
```

Comment:
A little bit of a hack here to make sure the macos window renders locally - not sure why it doesn't work for me if I don't do this.
```suggestion
        '<div id="termynal" data-termynal class="termy" data-ty-macos data-ty-lineDelay="100" data-ty-typeDelay="50" title="Cookiecutter Data Science">'
```